File: app/src/main/java/com/mdkashif/alarm/utils/AppConstants.java
package com.mdkashif.alarm.utils;

/**
 * Created by Kashif on 16-Apr-18.
 */
public class AppConstants {

    /////// Endpoints ///////
    private static String MAIN_URL = "http://www.mdkashif.com/UniversalAlarm/";
    public static String PRAYER_API = "http://api.aladhan.com/v1/";
    public static String FAQ= MAIN_URL+"faq.html";
    public static String PP= MAIN_URL+"privacy-policy.html";
    public static String TNC= MAIN_URL+"terms-and-conditions.html";

    /////// Miscellaneous ///////
    public final static String isFirstTimeLaunch = "isFirstTimeLaunch";
    public final static String ringStatus = "ring_status";
    public final static String ringtoneUri = "ringtone_uri";
    public final static String vibrateStatus = "vibrate_status";

    public final static String alarmTypeLocation = "location";
    public final static String alarmTypeBattery = "battery";
    public final static String alarmTypeTime = "time";
    public final static String alarmTypePrayer = "prayer";

    public final static String themeLight = "Light";
    public final static String themeDark = "Dark";
    public final static String themeSelected = "theme";

    public final static String city = "city";
    public final static String country = "country";

    /////// SetBatteryLevelFragment ///////
    public final static String highBatteryLevel = "hbl";
    public final static String lowBatteryLevel = "lbl";
    public final static String batteryTempLevel = "temp";
    public final static String batteryAlarmStatus = "batteryAlarmStatus";
    public final static String theftAlarmStatus = "theftAlarmStatus";

    /////// Notification ///////
    public final static Integer notificationId = 786;
    public final static String notificationChannelName = "Miscellaneous";
    public final static String notificationChannelId = "com.mdkashif.alarm.notifs";
    public final static String notificationDescription = "Alarms";

}

File: app/src/main/java/com/mdkashif/universalarm/utils/AppConstants.kt
package com.mdkashif.universalarm.utils

import com.mdkashif.universalarm.BuildConfig

/**
 * Created by Kashif on 16-Apr-18.
 */
object AppConstants {

    /////// Endpoints ///////
    private const val MAIN_URL = "http://www.mdkashif.com/UniversalAlarm/"
    const val GET_PRAYER_DATA_URL = "http://api.aladhan.com/v1/"
    const val FAQ = MAIN_URL + "faq.html"
    const val PP = MAIN_URL + "privacy-policy.html"
    const val TNC = MAIN_URL + "terms-and-conditions.html"

    /////// Miscellaneous ///////
    const val spAlias = "universal-alarm-sp"
    const val dbAlias = "universal-alarm-db"
    const val isFirstTimeLaunch = "isFirstTimeLaunch"
    const val ringtoneUri = "ringtone_uri"
    const val vibrateStatus = "vibrate_status"

    const val playServicesResolutionRequest = 1000
    const val updateInterval = 10000
    const val fastestInterval = 1000
    const val displacement = 1000

    const val city = "city"
    const val country = "country"

    const val snoozeTimeArrayPosition = "snoozeTimeArrayPosition"
    const val locationPrecisionArrayPosition = "locationPrecisionArrayPosition"
    const val timezone = "timezone"
    const val islamicDate = "islamicDate"
    const val islamicMonth = "islamicMonth"
    const val REQUEST_CODE_ENABLE_THEFT_ALARM = 121

    /////// SetBatteryLevelFragment ///////
    const val highBatteryLevel = "hbl"
    const val lowBatteryLevel = "lbl"
    const val batteryTempLevel = "temp"
    const val batteryAlarmStatus = "batteryAlarmStatus"
    const val temperatureAlarmStatus = "temperatureAlarmStatus"
    const val theftAlarmStatus = "theftAlarmStatus"
    const val theftPinEnabled = "theftPinEnabled"

    /////// Notification ///////
    const val notificationId = 786
    const val notificationChannelName = "Miscellaneous"
    const val notificationChannelId = BuildConfig.APPLICATION_ID + ".notifs"
    const val notificationDescription = "Alarms"
}
